stride_ws/src/robot_commander/scripts/robot_commander.py

```diff
@@ -82,18 +83,69 @@ def _send_velocity_command_using_radius(self, speed):
 
         self.velocity_command_publisher.publish(pose2d)
 
+    def _rate_limiter(self, speed_goal, acceleration, total_distance_in_index):
+        initial_time_in_s = time.time()
+        current_velocity = 0
+        is_starting_decel = True
+        frequency = 50 #hz
+        period = 1/frequency
+        r = rospy.Rate(frequency) 
+        minimum_decel_vel = 0.3 #m/s
+       
+        if speed_goal < 0: #reverse motion
+            acceleration_distance =  total_distance_in_index*2/3 
+            const_vel_distance = total_distance_in_index/3
+
+            while self.current_path_index > 0 and let_script_runs:
+                if self.current_path_index >=  acceleration_distance:
+                    velocity_input = math.copysign((min(abs(acceleration * (time.time()-initial_time_in_s)), abs(speed_goal))), speed_goal)
+                    self._send_velocity_command_using_radius(velocity_input)
+                    r.sleep()
+                elif self.current_path_index >=  const_vel_distance:
```

Comment:
Further verification: 
Consider the scenario path is 30m long, acceleration_distance = 20m from start, const_vel_distance = 10m from start
Robot start, and stop at 15m from start. What is the velocity_input if the robot returns to start?